util/metrics/import_metrics_files.rb

```diff
@@ -0,0 +1,181 @@
+      server_tag = "#{@prefix}_#{@id}_#{hostname}"
+      pattern = "'#{host_dir}/*.json'"
+      cmd = "ruby #{@json2graphite_path} --pattern #{pattern}" \
+        " --convert-to influxdb --netcat localhost --influx-db puppet_metrics --server-tag #{server_tag}"
+      puts "Importing puppet-metrics-collector files for host: #{hostname}"
+      puts " cmd: #{cmd}"
+      puts
+
+      `#{cmd}`
+    end
+  end
+end
+
+if $PROGRAM_NAME == __FILE__
+
+  DEFAULT_JSON2GRAPHITE_PATH = File.expand_path "~/git/puppet-metrics-viewer/json2graphite.rb"
+
+  DESCRIPTION = <<~DESCRIPTION
+    This script imports metrics collector data for the Standard and Large reference architectures.
+    It checks each service subdirectory of the puppet-metrics-collector directory and calls the json2graphite.rb script once for each host subdirectory found within.
+    The server tags are constructed using the following pattern:
+    prefix_id_hostname
+
+    For example:
+    be ruby util/metrics/import_metrics.rb -r results/scale/PERF_SCALE_12345 -p slv-649
+
+    This will result in the following server tags:
+    slv-649_12345_127.0.0.1
+    slv-649_12345_ip-10-227-0-22.amz-dev.puppet.net
+    slv-649_12345_ip-10-227-2-173.amz-dev.puppet.net
+
+  DESCRIPTION
+
+  DEFAULTS = <<~DEFAULTS
+
+    The following defaults values are used if the options are not specified:
+    * JSON2GRAPHITE_PATH (-j, --json2graphite): #{DEFAULT_JSON2GRAPHITE_PATH}
+
+  DEFAULTS
+
+  options = {}
+
+  # Note: looks like 'Store options to a Hash' doesn't work in Ruby 2.3.0.
+  # https://ruby-doc.org/stdlib-2.6.3/libdoc/optparse/rdoc/OptionParser.html
+  #  `end.parse!(into: options)`
+  # TODO: update to use '(into: options)' after Ruby update
+  OptionParser.new do |opts|
+    opts.banner = "Usage: import_metrics_files.rb [options]"
+
+    opts.on("-h", "--help", "Display the help text") do
+      puts DESCRIPTION
+      puts opts
+      puts DEFAULTS
+      exit
+    end
+
+    opts.on("-r", "--results_dir dir_path", String, "The results directory to process") do |results_dir|
+      options[:results_dir] = results_dir
+    end
+
+    opts.on("-p", "--prefix prefix", String, "The prefix to use when building the server tag") do |prefix|
+      options[:prefix] = prefix
+    end
+
+    opts.on("-j", "--json2graphite file_path", String, "The json2graphite script path") do |json2graphite|
+      options[:json2graphite] = json2graphite
+    end
+  end.parse!
+
+  raise "A results directory must be specified with the -r or --results_dir option" if options[:results_dir].nil?
+  raise "A prefix must be specified with the -p or --prefix option" if options[:prefix].nil?
+
+  raise "Specified directory does not exist: #{options[:results_dir]}" unless File.directory? options[:results_dir]
+
+  json2graphite_path = options[:json2graphite] || DEFAULT_JSON2GRAPHITE_PATH
+  raise "The json2graphite.rb script was not found: #{json2graphite_path}" unless File.exist? json2graphite_path
```

Comment:
if you are going to call it with ruby, maybe should check if ruby is on the path here to avoid getting a sea of errors for each attempt to run the script with ruby.

Reply:
Updated to remove `ruby` from cmd.